src/typedefs.jl

```diff
@@ -2,105 +2,54 @@ typealias DefaultRefType UInt32
 
 ## Pools
 
-abstract CategoricalPool{T, R <: Integer, V}
-
 # V is always set to NominalValue{T} or OrdinalValue{T}
 # This workaround is needed since this type not defined yet
 # See JuliaLang/julia#269
-for P in (:NominalPool, :OrdinalPool)
-    @eval begin
-        immutable $P{T, R <: Integer, V} <: CategoricalPool{T, R, V}
-            index::Vector{T}
-            invindex::Dict{T, R}
-            order::Vector{R}
-            ordered::Vector{T}
-            valindex::Vector{V}
-
-            function $P{T, R}(index::Vector{T},
-                              invindex::Dict{T, R},
-                              order::Vector{R})
-                pool = new(index, invindex, order, index[order], V[])
-                buildvalues!(pool)
-                pool
-            end
-        end
+immutable CategoricalPool{T, R <: Integer, V, O}
```

Comment:
Having "ordered" be a type parameter means it's still necessary to change the type to move from nominal to ordinal. I was thinking of making it a boolean field of `CategoricalPool` and `CategoricalValue` instead.